lib/vssh/libssh.c

```diff
@@ -2167,35 +2168,67 @@ static CURLcode myssh_connect(struct connectdata *conn, bool *done)
     return CURLE_FAILED_INIT;
   }
 
-  ssh_options_set(ssh->ssh_session, SSH_OPTIONS_FD, &sock);
+  rc = ssh_options_set(ssh->ssh_session, SSH_OPTIONS_HOST, conn->host.name);
+  if(rc != SSH_OK) {
+    failf(data, "Could not set remote host");
```

Comment:
If you don't consider this a "lethal" problem that makes you return an error, you should rather use `infof()` instead of `failf()`. The difference is basically that the latter stores the message in the error buffer.

But really, shouldn't these failures rather be a reason to return error? They will seriously change what happens and will (if passed through) do something other than what the user asked for.

Reply:
Sorry, I misunderstood the purpose of ``failf()``, the idea was to keep the previous behaviour and add the missing return code checking to log eventual issues.

I think it should fail for all required arguments and ignore the optional ones (but add a message to the log in case of failure). I'll change the code and push again.